### ISSUE TITLE
Plan has not been validated. See reason (s) above.

### ISSUE THREAD

**Issue by bingws** [HewlettPackard/dlcookbook-dlbs#8]
I tried the following command, but got no result.
```
python ./python/dlbs/experimenter.py run\
             -Pexp.framework='"tensorflow"'\
             -Pexp.gpus='"0"'\
             -Vexp.docker=true\
             -Pexp.log_file='"./benchmarks/my_experiment/tf_${exp.model}_${exp.replica_batch}.log"'\
             -Vexp.model='["alexnet", "googlenet"]'\
             -Vexp.replica_batch='[2, 4]'\
             -Ptensorflow.docker_image='"hpe/tensorflow:cuda9-cudnn7"'
```
```
WARNING:root:Module 'matplotlib' cannot be imported, certain system information will not be available
INFO:root:Plan was built with 4 experiments
Plan was built with 4 experiments
====================== VALIDATION REPORT =======================
=========================== MESSAGES ===========================
[
    {
        "check_name": "CanRunDocker", 
        "output": "Docker version 18.09.0, build 4d60db4\n", 
        "cmd": "nvidia-docker --version", 
        "retcode": 0
    }
]
======================== FRAMEWORK STATS =======================
{
    "tensorflow": {
        "num_disabled": 0, 
        "docker_images": [
            "hpe/tensorflow:cuda9-cudnn7"
        ], 
        "num_exps": 4, 
        "num_cpu_exps": 0, 
        "num_gpu_exps": 4, 
        "num_host_exps": 0, 
        "num_docker_exps": 4
    }
}
============================ ERRORS ============================
Other errors:
[
    {
        "check_name": "DockerImageExists", 
        "output": "[]\n Error: No such image: hpe/tensorflow:cuda9-cudnn7\n", 
        "cmd": "docker inspect --type=image hpe/tensorflow:cuda9-cudnn7", 
        "retcode": 1
    }
]
========================= PLAN SUMMARY =========================
Is plan OK ................................ False
Total number of experiments (plan size).....4
Number of disabled experiments ............ 0
Number of active experiments .............. 4
Log files collisions ...................... NO
================================================================
WARNING:root:Plan has not been validated. See reason (s) above.
Plan has not been validated. See reason (s) above.
WARNING:root:If you believe validator is wrong, rerun experimenter with `--no-validation` flag.
If you believe validator is wrong, rerun experimenter with `--no-validation` flag.

```

My PC environment is as follows:
Ubuntu16.04
Docker version 18.09.0
NVIDIA 1080Ti

$ docker images:
```
REPOSITORY                           TAG                 IMAGE ID            CREATED             SIZE
registry.docker-cn.com/nvidia/cuda   latest              1cc6f1613121        2 weeks ago         2.24GB
nvcr.io/nvidia/tensorflow            18.07-py3           8289b0a3b285        5 months ago        3.34GB
```

**Comment by sergey-serebryakov** [comment]
@bingws You need to provide an `exp.docker_image` parameter to specify your own docker image:
```bash
-Pexp.docker_image='"nvcr.io/nvidia/tensorflow:18.07-py3"'
```
TensorFlow benchmark backend (`tensorflow`) is actually a Google's tf_cnn_benchmark project. We noticed that it did not scale very nicely on 8 GPU servers. I recommend using `nvtfcnn` benchmark backend. It's an optimized benchmark project from NVIDIA for TensorFlow.  You can do this by providing the following parameter:
```bash
-Pexp.framework='"nvtfcnn"'
```
Default docker image for this benchmark backend is exactly the one you have in your system [nvcr.io/nvidia/tensorflow:18.07-py3](https://github.com/HewlettPackard/dlcookbook-dlbs/blob/master/python/dlbs/configs/nvtfcnn.json#L21).
Also, please make sure you have nvidia-docker plugin installed.

BTW, we have updated the documentation pages and now they use docker images from NGC in provided [examples](https://hewlettpackard.github.io/dlcookbook-dlbs/#/?id=quick-start).

**Comment by bingws** [comment]
Thank you very  much.I was first exposed to GPU performance testing, so I didn't quite understand.I modified the command according to your suggestion and got the following results.
```
WARNING:root:Module 'matplotlib' cannot be imported, certain system information will not be available
INFO:root:Plan was built with 4 experiments
Plan was built with 4 experiments
INFO:root:Plan has been validated
Plan has been validated
--------------------------------------------------------------
Experimenter pid 5322. Run this to gracefully terminate me:
        kill -USR1 5322
I will terminate myself as soon as current benchmark finishes.
--------------------------------------------------------------
__launcher.start_time__="2018-11-30 21:11:23.669829"
__launcher.total_experiments__=4
__launcher.skipped_experiments__=0
__launcher.failed_experiments__=0
__launcher.hours__=0.027950213888888888
__launcher.disabled_experiments__=0
__launcher.end_time__="2018-11-30 21:13:04.290599"
__launcher.active_experiments__=4
```
One more question I want to ask: is this __launcher. Hours__ =0.027950213888888888 only for gpus?There is no way to compare, for example, with a CPU or multiple gpus.

**Comment by sergey-serebryakov** [comment]
@bingws You have only one GPU, so `'"0,0"'` does not look like a right parameter value. Just use `-Pexp.gpus='"0"'`. The TensorFlow `nvtfcnn` benchmark backend for each model uses its own startup file. The error you see (`can't open file '.py'`) means that somehow DLBS did not pick the right model name and used empty string instead.
Please, study the following script with brief description. I just ran it and got benchmark results:
```bash
# Make sure you have the docker image `nvcr.io/nvidia/tensorflow:18.07-py3`. To get it, run:
#     docker pull nvcr.io/nvidia/tensorflow:18.07-py3
docker images

# Go to DLBS root folder
cd ~/projects/dlbs

# Initialize host environment (export some variables)
source ./scripts/environment.sh 

# Remove old log files
[ -d "./benchmarks/my_experiment" ] && rm -f "./benchmarks/my_experiment"

# Run benchmarks. The nvtfcnn implements a version of AlexNet that's called AlexNetOWT.
# By default, DLBS uses docker to run benchmarks and uses `nvcr.io/nvidia/tensorflow:18.07-py3` image,
#      so, I am not providing those parameters on a command line
python $experimenter run -Pexp.framework='"nvtfcnn"' -Pexp.gpus='"0"' -Vexp.model='["googlenet", "alexnet_owt"]'  -Vexp.replica_batch='[2, 4]' -Pexp.log_file='"./benchmarks/my_experiment/tf_${exp.model}_${exp.replica_batch}.log"'
```

Once DLBS runs all four benchmarks, the `./benchmarks/my_experiment` folder should contain 4 files with `log` extension. Each file will contain a `results.throughput` parameter with achieved performance (images/sec), for instance:
```bash
cat ./benchmarks/my_experiment/tf_googlenet_4.log | grep results.throughput
__results.throughput__=150.766667
```

DLBS provides some basic post-processing capabilities, so you can get some summary of done benchmarks automatically:
```bash
# Parse log files and create a JSON summary of all benchmarks:
 python $logparser "./benchmarks/my_experiment/" --output_file "./benchmarks/my_experiment/results.json"

# Print a user friendly summary of those results:
python $reporter --summary_file './benchmarks/my_experiment/results.json' --type 'exploration' --target_variable 'results.time'
```

When running the last command, you should expect seeing something like this:
```bash
Batch time (milliseconds)
Network              Device     2          4
AlexNetOWT           0          12.03      12.45
GoogleNet            0          26.43      26.53



Inferences Per Second (IPS, throughput)
Network              Device     2          4
AlexNetOWT           0          166        321
GoogleNet            0          75         150
```
If you want to run GPU and CPU workloads, this `$reporter` script will probably fail (it was tested with GPUs), so, you may need to examine log files manually. Search for `results.throughput` parameter in those log files as I did above (with `grep` tool).

Some comments:
1. Batch sizes of 2 and 4 are probably way to small and should not be used for benchmarking a training phase. In general, for inference benchmarks this is probably OK.
2. You really want to provide a `exp.num_batches` parameter that specifies number of benchmark iterations. Default value of 100 is small. Something like 500 iterations is a good starting point: `-Pexp.num_batches=500`.

Basically, I followed the script presented [here](https://hewlettpackard.github.io/dlcookbook-dlbs/#/?id=quick-start).

**Comment by bingws** [comment]
@sergey-serebryakov Thank you very much！
